### DISCUSSION TITLE
CppTools extension burning CPU

### DISCUSSION THREAD

**Discussion by fhmlopez** [microsoft/vscode-cpptools#8133] [General]
Has any body notice when opening VSCode the `cpptools` extension starts burning CPU like crazy?

Running `htop` shows CPU% 99.9 for a process running from `~/.vscode/extensions/ms-vscode.cpptools-1.6.0/bin/cpptools`

this is the VSCode version I am using

> Version: 1.60.0
> Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
> Date: 2021-09-01T10:43:02.136Z
> Electron: 13.1.8
> Chrome: 91.0.4472.164
> Node.js: 14.16.0
> V8: 9.1.269.39-electron.0
> OS: Linux x64 5.11.0-34-generic

**Comment by bobbrow** [comment]
When you open a folder for the first time, there is a bit of indexing/parsing of source files that happens so that the language features work.  The amount of time it takes depends on the size of your workspace, but you can track the progress by hovering over the database icon in the status bar.  If that icon is gone and you continue to see 99% CPU, you should open a bug and we can start an investigation into what is wrong.

If you prefer the extension to not use the CPU aggressively, you can change the `C_Cpp.workspaceParsingPriority` setting.
